internal/pkg/template/repository/manifest/manifest.go

```diff
@@ -70,29 +70,33 @@ func (m *Manifest) Save(fs filesystem.Fs) error {
 	return nil
 }
 
-func (m *Manifest) Get(templateId string) (*TemplateRecord, bool) {
-	v, ok := m.records[templateId]
-	return v, ok
+func (m *Manifest) IsChanged() bool {
+	return m.changed
 }
 
-func (m *Manifest) GetOrCreate(templateId string) *TemplateRecord {
-	record, found := m.Get(templateId)
-	if !found {
-		record = newRecord(templateId)
-		m.add(record)
+func (m *Manifest) Persist(records ...TemplateRecord) {
+	for _, record := range records {
+		m.records[record.Id] = record
+		m.changed = true
```

Comment:
Narazil som na to, ze potrebujeme vediet ci sa `repository manifest` zmenil (a je potrebne ho ulozit).
Preto som pridal metodu `Persist` + spravil records immutable (odstranil pointre).
Takto sa z `repository manifest` vzdy vrati kopia `TemplateRecord`, a ak sa record zmeni, tak sa musi jeho zmena ulozit pomocou `Persist`.